exaudfclient/base/script_options_parser/ctpg/test/script_option_lines_test.cpp

```diff
@@ -0,0 +1,214 @@
+#include "base/script_options_parser/ctpg/script_option_lines_ctpg.h"
+#include <gtest/gtest.h>
+#include <string>
+#include <exception>
+
+const std::string lineEnd = ";";
+
+class TestException : public std::runtime_error {
+    using std::runtime_error::runtime_error;
+};
+
+void throwException(const char* ex) {
+    throw TestException(std::string(ex));
+}
+
+using namespace ExecutionGraph::OptionsLineParser::CTPG;
+
+
+class ScriptOptionLinesWhitespaceTest : public ::testing::TestWithParam<std::tuple<std::string, std::string, std::string, std::string, std::string, std::string>> {};
```

Comment:
Not possible here:
Using const std::string would require to have std::vector<const string> below. But that's not allowed  because:

>  The actual problem is that const std::string is not [Destructible](http://en.cppreference.com/w/cpp/concept/Destructible),

 see https://stackoverflow.com/a/21365478